src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -151,5 +152,27 @@ private void CalculateSearchParameterHash()
                     (resourceType, existingValue) => searchParamHash);
             }
         }
+
+        public void DeleteSearchParameter(ITypedElement searchParam)
+        {
+            var searchParamWrapper = new SearchParameterWrapper(searchParam);
+
+            if (!UrlLookup.Remove(new Uri(searchParamWrapper.Url)))
+            {
+                throw new Exception(string.Format(Resources.CustomSearchParameterNotfound, searchParamWrapper.Url));
+            }
+
+            var searchParameterInfo = TypeLookup[searchParamWrapper.Type][searchParamWrapper.Name];
+            var allResourceTypes = searchParameterInfo.TargetResourceTypes.Union(searchParameterInfo.BaseResourceTypes);
+            foreach (var resourceType in allResourceTypes)
+            {
+                if (TypeLookup.ContainsKey(resourceType))
+                {
+                    TypeLookup[resourceType].Remove(searchParameterInfo.Name);
```

Comment:
>.Remove [](start = 44, length = 7)

How we protect ourselves from concurrency issues?
From what I can see in code, that `IDictionary<string, SearchParameterInfo>` is just a regular Dictionary, and it's ok if you instantiate it once during startup and just do lookup for it, but now we adding and removing elements from it, and I'm not sure is it locked for one thread in upstream or just not locked at at all.